base/task.jl

```diff
@@ -1184,7 +1194,16 @@ checktaskempty = Partr.multiq_check_empty
 @noinline function poptask(W::StickyWorkqueue)
     task = trypoptask(W)
     if !(task isa Task)
-        task = ccall(:jl_task_get_next, Ref{Task}, (Any, Any, Any), trypoptask, W, checktaskempty)
+        process_events()
+        task = trypoptask(W)
+    end
+    if !(task isa Task)
+        sched_task = get_sched_task()
+        if current_task() === sched_task
+            task = ccall(:jl_task_get_next, Ref{Task}, (Any, Any, Any), trypoptask, W, checktaskempty)
+        else
+            yieldto(sched_task)
```

Comment:
A `@goto` the beginning of `poptask` here resulted in a freeze while building:
```
Random  ────────  1.407750 seconds
Stdlibs total  ─ 20.005079 seconds
Sysimage built. Summary:
Base ────────  58.894729 seconds 74.6422%
Stdlibs ─────  20.005079 seconds 25.3541%
Total ───────  78.902671 seconds
^C
```
🤔

Reply:
Yes, if the first thing a Task does after getting scheduled is to go back to sleep, nothing will ever get done

Reply:
You are missing an assignment of task=current_task() here